packages/bridge-ui/src/proof/service.spec.ts

```diff
@@ -58,6 +73,38 @@ describe("prover tests", () => {
     block.baseFeePerGas = "0";
   });
 
+  it("throws on invalid proof", async () => {
+    mockProvider.send.mockImplementation(
+      (method: string, params: unknown[]) => {
+        if (method === "eth_getBlockByNumber") {
+          return block;
+        }
+
+        if (method === "eth_getProof") {
+          return invalidStorageProof;
+        }
+      }
+    );
+
+    const srcChain = 167001;
+    const map = new Map<number, ethers.providers.JsonRpcProvider>();
+    map.set(
+      srcChain,
+      mockProvider as unknown as ethers.providers.JsonRpcProvider
```

Comment:
nit: possible to make this more type safe? maybe we can declare at the top as: `Pick<ethers.providers.JsonRpcProvider, "send">`, and then we can remove the type assertion here. and maybe we can just import JsonRpcProvider instead of all of ethers? all good if thats not possible. also looks like BigNumber is unused from ethers.

Reply:
or rather declare the map as such is what i mean. and at the top declaration.

Reply:
the map is a dependency, `as unknown as {type}` is a way to add a mock type but only implement the method we want. Can you push up your implementation?